common/storage/sql/mariadb/storage.c

```diff
@@ -7,13 +7,394 @@
+    return RC_STORAGE_FAILED_BINDING;
+  }
+
+  if (mysql_stmt_execute(mariadb_statement) != 0) {
+    log_statement_error(mariadb_statement);
+    return RC_STORAGE_FAILED_EXECUTE;
+  }
+
+  transaction_reset(transaction);
+
+  if (model == MODEL_TRANSACTION) {
+    storage_transaction_load_bind_essence(bind_out, transaction, &index);
+    storage_transaction_load_bind_attachment(bind_out, transaction, &index);
+    storage_transaction_load_bind_consensus(bind_out, transaction, &index);
+    storage_transaction_load_bind_data(bind_out, transaction, &index);
+  } else if (model == MODEL_TRANSACTION_ESSENCE_METADATA) {
+    storage_transaction_load_bind_essence(bind_out, transaction, &index);
+    storage_transaction_load_bind_metadata(bind_out, transaction, &index);
+  } else if (model == MODEL_TRANSACTION_ESSENCE_ATTACHMENT_METADATA) {
+    storage_transaction_load_bind_essence(bind_out, transaction, &index);
+    storage_transaction_load_bind_attachment(bind_out, transaction, &index);
+    storage_transaction_load_bind_metadata(bind_out, transaction, &index);
+  } else if (model == MODEL_TRANSACTION_ESSENCE_CONSENSUS) {
+    storage_transaction_load_bind_essence(bind_out, transaction, &index);
+    storage_transaction_load_bind_consensus(bind_out, transaction, &index);
+  } else if (model == MODEL_TRANSACTION_METADATA) {
+    storage_transaction_load_bind_metadata(bind_out, transaction, &index);
+  } else {
+    return RC_STORAGE_FAILED_NOT_IMPLEMENTED;
+  }
+
+  if (mysql_stmt_bind_result(mariadb_statement, bind_out) != 0) {
+    log_statement_error(mariadb_statement);
+    return RC_STORAGE_FAILED_BINDING;
+  }
+
+  if (mysql_stmt_store_result(mariadb_statement) != 0) {
+    log_statement_error(mariadb_statement);
+    return RC_STORAGE_FAILED_STORE_RESULT;
+  }
+
+  pack->num_loaded = mysql_stmt_num_rows(mariadb_statement);
+  pack->insufficient_capacity = pack->num_loaded > pack->capacity;
+
+  if (pack->num_loaded == 0 || pack->insufficient_capacity == true) {
+    return RC_OK;
+  }
+
+  if (mysql_stmt_fetch(mariadb_statement) != 0) {
+    log_statement_error(mariadb_statement);
+    return RC_STORAGE_FAILED_STORE_RESULT;
+  }
+
+  return RC_OK;
+}
+
+static retcode_t storage_milestone_load_generic(MYSQL_STMT* const mariadb_statement, MYSQL_BIND* const bind_out,
+                                                iota_stor_pack_t* const pack) {
+  iota_milestone_t* milestone = *((iota_milestone_t**)pack->models);
+
+  if (mysql_stmt_execute(mariadb_statement) != 0) {
+    log_statement_error(mariadb_statement);
+    return RC_STORAGE_FAILED_EXECUTE;
+  }
+
+  milestone_reset(milestone);
+
+  bind_out[0].buffer = (char*)&milestone->index;
+  bind_out[0].buffer_type = MYSQL_TYPE_LONGLONG;
+
+  bind_out[1].buffer = (char*)milestone->hash;
+  bind_out[1].buffer_type = MYSQL_TYPE_BLOB;
+  bind_out[1].buffer_length = FLEX_TRIT_SIZE_243;
+
+  if (mysql_stmt_bind_result(mariadb_statement, bind_out) != 0) {
```

Comment:
can you group both `mysql_stmt_bind_result` `mysql_stmt_store_result` into a *bind_and_store function?